### PR TITLE
Create time log contract

### PR DESCRIPTION
**Pull request by xaler5**

### CODE REVIEW

File: contracts/TimeLogger.sol [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    {
+        uint today = time / SECONDS_PER_DAY;
+
+        _setDay(today);
+
+        if (lastOpenDay != 0) {
+            if (today > lastOpenDay) {
+                _completePendingDays(today, time, source);
+            } else {
+                _completeLastLogOfDay(today, time, source);
```
Comment (lales12) [inline contracts/TimeLogger.sol]: Why dont user close log?

File: contracts/TimeLogger.sol [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+        uint thisDay,
+        uint endTime
+    )
+        internal
+    {
+        uint index = day[thisDay].timelogs[day[thisDay].timelogs.length - 1];
+
+        timeLog[index].timeEnd = endTime;
+
+        timeLog[index].valid = false;
```
Comment (lales12) [inline contracts/TimeLogger.sol]: Why is not valid?

File: contracts/TimeLogger.sol [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+        internal
+    {
+        timeLog.push(TimeLog(startTime, 0, source, true));
+
+        day[thisDay].timelogs.push(timeLog.length - 1);
+
+        _publish();
+    }
+
+    function _closeLog(
```
Comment (lales12) [inline contracts/TimeLogger.sol]: this function can be more reusable

File: contracts/TimeLogger.sol [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+        uint today,
+        uint lastDay
+    )
+        internal
+    {
+        //fulfill the days in the gap with 24h invalid single log
+        for (uint i = lastDay + 1; i < today; i++) {
+            uint[] memory tmpArray;
+
+            timeLog.push(TimeLog(0, 0, SOLIDITY_SOURCE, false));
```
Comment (lales12) [inline contracts/TimeLogger.sol]: reuse the functions create log and close log

File: contracts/User.sol [hunk 5]
```diff
@@ -184,26 +190,20 @@ contract User is UserInterface {
...
-        uint createdAt
+    function clauseNotification(
+        address clauseContract,
+        string memory clauseType,
+        string memory notificationType,
+        string memory id
     )
         public
-        signaturitOnly
+        onlyOwner
```
[the remark this answers is not in the document]
Comment (xaler5) [reply contracts/User.sol]: fixed and changed for onlyAllowed

**Pull request closed without merging**